docs/technical-design/launch-darkly-testing-approach.md

```diff
@@ -1,15 +1,149 @@
+
+```typescript
+  function testLDService(
+      mockFeatureFlags?: Partial<FeatureFlagObject>
+  ): LDService {
+      const featureFlags = defaultFeatureFlags
+
+      //Update featureFlags with mock flag values.
+      if (mockFeatureFlags) {
+        for (const flag in mockFeatureFlags) {
+          const featureFlag = flag as FeatureFlagTypes
+          featureFlags[featureFlag] = mockFeatureFlags[
+                  featureFlag
+                  ] as FlagValueTypes
+        }
+      }
+
+      return {
+        getFeatureFlag: async (user, flag) => featureFlags[flag],
+      }
+  }
+```
+
+We then pass `testLDService` with our defined flag values into `constructTestPostgresServer` when setting up test server. Resolvers in the test will now use `testLDService` when calling `getFeatureFlag` returning our defined feature flag values.
+
+```javascript
+it('does not error when risk based question is undefined and rate-cert-assurance feature flag is off', async () => {
+    const mockLDService = testLDService({'rate-cert-assurance': false})
+    const server = await constructTestPostgresServer({ 
+      ldService: mockLDService,
+    })
+
+    // setup
+    const initialPkg = await createAndUpdateTestHealthPlanPackage(server, {
+      riskBasedContract: undefined,
+    })
+  
+    ...
+})
+```
+
 ## Feature Flag Cypress Testing
 Currently, there is no out of the box Cypress integration with LaunchDarkly. Our implementation approach enables the testing of multiple flag values independent of what is set in LaunchDarkly by intercepting api calls and returning our own generated flag values. This allows us to dynamically tests UI in Cypress with different flag values without having to modify flag values in the LaunchDarkly dashboard.
 
-The implementation is heavily relent on the data and generated Types in `flags.ts` located in `app-web/src/common-code/featureFlags`. This is done for several reasons:
-- Centralize feature flag lists so when adding new flags it only needs to be done once.
-- Auto generated Types based on the feature flag list help removes the need to update types manually and throughout the codebase.
-- Autocompletion of flag parameters to prevent mistakes and invalid flags when using custom LaunchDarkly Cypress commands.
-- Type checking in custom LaunchDarkly Cypress commands.
-
-Overall this improves development time of UI tests behind feature flags.
+### Limitations
+There is one major limitation to this implementation, server and client side flag value syncing in Cypress runs. Since we are not reaching out to LaunchDarkly to either retrieve or set a flag value the server side does not know we have changed a flags value. In this scenario the same flag would be in different states between client and server. 
```

Comment:
```suggestion
There is one major limitation to this implementation: server and client side flag value syncing in Cypress runs. Since we are not reaching out to LaunchDarkly to either retrieve or set a flag value the server side does not know we have changed a flag's value. In this scenario the same flag would be in different states between client and server. 
```